### PR TITLE
Made environment and compounds panel collapsed state be saved

### LINKED ISSUE #3320: Make environment panel collapsed state be remembered
so that going to the editor can coming back doesn't result in the state being reset and requiring to press it again

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Made their collapsed states savable, refactored the expand/collapsed logic a bit while working. Also shouldn't their collapsed state have already been kept after returning from the editor since past versions?

**Related Issues**

Fixes #3320.

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Review by RacerBG — APPROVED** [review]
I had never thought about this, it's nice to have this QoL fix.

File: src/microbe_stage/MicrobeStage.tscn [hunk 1]
```diff
@@ -2249,11 +2253,12 @@ custom_styles/separator = SubResource( 36 )
 anims/FadeInOut = SubResource( 37 )
 
 [node name="PauseMenu" parent="." instance=ExtResource( 11 )]
+visible = true
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeStage.tscn:2256]: Unintended change?

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
I guess it's fine if it is saved in the save files, though I'm not sure if that was what was the intention with the original issue?
Anyway I can accept the saving. I spotted one potentially incorrect visibility change, so that should be confirmed it is intended before merging.

File: src/microbe_stage/MicrobeStage.tscn (hunk 1 above)
Comment (Kasterisk) [reply src/microbe_stage/MicrobeStage.tscn]: Probably. Now fixed.

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**